server/src/main/java/org/apache/cassandra/sidecar/job/RepairJob.java

```diff
@@ -0,0 +1,249 @@
+import org.apache.cassandra.sidecar.common.request.data.RepairPayload;
+import org.apache.cassandra.sidecar.common.server.StorageOperations;
+import org.apache.cassandra.sidecar.config.RepairConfiguration;
+import org.apache.cassandra.sidecar.handlers.data.RepairRequestParam;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Implementation of {@link OperationalJob} to perform repair operation.
+ */
+public class RepairJob extends OperationalJob
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(RepairJob.class);
+    private static final String OPERATION = "repair";
+    private static final String PREVIEW_KIND_REPAIRED = "REPAIRED";
+
+    private final RepairRequestParam repairParams;
+    private final Vertx vertx;
+    private final RepairConfiguration repairConfiguration;
+    protected StorageOperations storageOperations;
+
+    /**
+     * Enum representing the status of a parent repair session
+     */
+    public enum ParentRepairStatus
+    {
+        IN_PROGRESS, COMPLETED, FAILED
+    }
+
+    /**
+     * Constructs a job with a unique UUID, in Pending state
+     *
+     * @param vertx
+     * @param repairConfiguration
+     * @param jobId               UUID representing the Job to be created
+     * @param storageOps
+     * @param repairParams
+     */
+    public RepairJob(Vertx vertx, RepairConfiguration repairConfiguration, UUID jobId, StorageOperations storageOps, RepairRequestParam repairParams)
+    {
+        super(jobId);
+        this.vertx = vertx;
+        this.repairConfiguration = repairConfiguration;
+        this.storageOperations = storageOps;
+        this.repairParams = repairParams;
+    }
+
+    @Override
+    public boolean isRunningOnCassandra()
+    {
+        // TODO: Leverage repair vtables to fail-fast on conflicting repairs (overlapping token-ranges or replica-sets)
+        // Currently does not check for concurrent repairs
+        return false;
+    }
+
+    @Override
+    protected void executeInternal()
+    {
+        Map<String, String> options = generateRepairOptions(repairParams.requestpayload());
+        String keyspace = repairParams.keyspace().name();
+
+        LOGGER.info("Executing repair operation for keyspace {} jobId={} maxRuntime={}",
+                    keyspace, this.jobId(), repairConfiguration.maxRepairJobRuntimeMillis());
+
+        int cmd = storageOperations.repair(keyspace, options);
+        if (cmd <= 0)
+        {
+            // repairAsync can only return 0 for replication factor 1.
+            LOGGER.info("Replication factor is 1. No repair is needed for keyspace '{}'", keyspace);
+        }
+        else
+        {
+            // complete the max wait time promise either when exceeding the wait time, or the result is available
+            Promise<Boolean> maxWaitTimePromise = Promise.promise();
+            vertx.setTimer(repairConfiguration.maxRepairJobRuntimeMillis(), d -> {
+                LOGGER.info("Timer Poll");
```

Comment:
Can add job ID and keyspace to the log message so that job can be traced in the log messages?